docs/src/DeveloperGuide/LogicSeparation.md

```diff
@@ -0,0 +1,493 @@
+block, a `quantum` program block, a `readout` block and a `post-classical`
+processing block. Visualised in as above the blocks are executed as follows:
+
+```text
+    Classical             PPU-SPU             Quantum
+ processing unit  │      interface      │ processing unit
+                  │                     │
+           entry│ │                     │
+                ▼ │        load         │
+                  ├─────────────────────▶
+                  │       readout       │ │ quantum
+                  ◀─────────────────────┤ ▼
+ post-classical │ │                     │
+                ▼ │                     │
+  exit-grouping │ │                     │
+                ▼ │                     │
+                  ▼                     ▼
+                  Time
+```
+
+With this separation it becomes straight-forward to execute the program parts in
+accordance with the processing unit which the require as well as to ensure that
+proper setup of the QPU was performed before initiating the quantum calculation.
+
+### Block Separation Strategy
+
+Our block separation strategy follows a divide-and-conquer approach: First we
+sort instructions into new blocks according to the source processing. Each of
+these blocks are then split into blocks based on the destination processing
+unit. We illustrate this in the following diagram:
+
+```text
+                                 ┌──────────┐
+                                 │Classical │     entry
+                 ┌──────────┐ ┌─▶│   Dest   │
+                 │          │ │  └──────────┘
+                 │Classical │ │  ┌──────────┐
+              ┌─▶│  Source  ├─┤  │ Quantum  │      load
+              │  │          │ └─▶│   Dest   │
+              │  └──────────┘    └──────────┘
+              │                  ┌──────────┐
+┌──────────┐  │  ┌──────────┐    │ Quantum  │    quantum
+│          │  │  │          │ ┌─▶│   Dest   │
+│ Original │  │  │ Quantum  │ │  └──────────┘
+│  block   │──┼─▶│  Source  ├─┤  ┌──────────┐
+│          │  │  │          │ │  │Classical │    readout
+└──────────┘  │  └──────────┘ └─▶│   Dest   │
+              │                  └──────────┘
+              │  ┌──────────┐    ┌──────────┐
+              │  │          │    │Classical │ post-classical
+              │  │Classical │ ┌─▶│   Dest   │
+              └─▶│  Source  │─┤  └──────────┘
+                 │          │ │  ┌ ─ ─ ─ ─ ─
+                 └──────────┘ │   Empty (QD)│
+                              └─▶│
+                                  ─ ─ ─ ─ ─ ┘
+```
+
+We note the second split is reversed depending type of instructions it contain.
+That is the destination of same type as the source processing unit comes first
+and the case where they are opposite follows. We also note that the second split
+where we divide depending on destination is technically easier to implement than
+the first since this the blocks processed already have certain guarantees
+provided:
```

Comment:
These two sentences are grammatically faulty. I don't know what the intended message is for these.

Reply:
Updated.